app/src/main/java/com/brandon/forzenbook/viewmodels/LoginViewModel.kt

```diff
@@ -0,0 +1,291 @@
+                                    location = location,
+                                    dateOfBirth = date,
+                                )
+                        }
+                        CreateUserOutcome.CREATE_USER_DUPLICATE -> {
+                            CreateAccountUiState.Idle(
+                                email = email,
+                                firstName = firstName,
+                                lastName = lastName,
+                                location = location,
+                                dateOfBirth = date,
+                                isDuplicateUser = true,
+                            )
+                        }
+                        CreateUserOutcome.CREATE_USER_SUCCESS -> {
+                            createAccountState.value = CreateAccountUiState.Loaded
+                        }
+                    }
+                }
+            } else {
+                createAccountState.value = CreateAccountUiState.Idle(
+                    isEmailError = validMail,
+                    isLocationError = validLoc,
+                    isDateError = validDob,
+                    email = email,
+                    firstName = firstName,
+                    lastName = lastName,
+                    location = location,
+                    dateOfBirth = date,
+                )
+            }
+        }
+        Log.e(VIEWMODEL_ERROR_TAG, "${createAccountState.value}")
+    }
+
+    private fun isValidLocation(location: String): Boolean {
+        return location.length <= LOCATION_CHAR_LIMIT
+    }
+
+    private fun isValidEmail(email: String): Boolean {
+        return PatternsCompat.EMAIL_ADDRESS.matcher(email).matches()
+    }
+
+    private fun isValidDateOfBirth(dob: String): Boolean {
+        dob.forEach {
+            if (!it.isDigit()) {
+                return false
+            }
+        }
+        return try {
+            val inputFormat = SimpleDateFormat("MMddyyyy")
+            val tempDate = inputFormat.parse(dob)
+            val outputFormat = SimpleDateFormat("yyyy-MM-dd")
+            val dateString = tempDate?.let { outputFormat.format(it) }
+            sqlDate = Date.valueOf(dateString)
+            true
+        } catch (e: Exception) {
+            Log.e(VIEWMODEL_ERROR_TAG, "Invalid Date of Birth Provided.")
+            false
+        }
+    }
+
+    private fun isValidCode(code: String): Boolean {
+        return code.isDigitsOnly()
+    }
+
+    fun isNetworkAvailable(context: Context?): Boolean {
+        if (context == null) return false
+        val connectivityManager =
+            context.getSystemService(Context.CONNECTIVITY_SERVICE) as ConnectivityManager
+        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.Q) {
+            val capabilities =
+                connectivityManager.getNetworkCapabilities(connectivityManager.activeNetwork)
+            if (capabilities != null) {
+                return true
+            }
+        } else {
```

Comment:
If there any reason to not just use this else block? Do you get anything out of the Q level methods?

Reply:
I suppose I don't, using the other version for sdk 29 or higher doesn't seem to do anything for me since the fallback check is older. Removed that half of the if statement.